test/types/marked.ts

```diff
@@ -246,21 +261,25 @@ marked.use(asyncExtension);
 const md = '# foobar';
 const asyncMarked: string = await marked(md, { async: true });
 const promiseMarked: Promise<string> = marked(md, { async: true });
-// @ts-expect-error marked can still be async if an extension sets `async: true`
-const notAsyncMarked: string = marked(md, { async: false });
-// @ts-expect-error marked can still be async if an extension sets `async: true`
+const notAsyncMarked: string = marked(md, { async: false, silent: true });
 const defaultMarked: string = marked(md);
-// as string can be used if no extensions set `async: true`
-const stringMarked: string = marked(md) as string;
```

Comment:
Marked calls that don't specify an async option should be `string | Promise<string>` since they could be either and they won't throw an error.

Reply:
> since they could be either

Just curious, how’s that? If no `async` is specified, it defaults to `false`, right?

Reply:
If no async is specified it should default to whatever async is set by extensions.

Reply:
```js
const syncMarked = new Marked({ async: false });
syncMarked.parse(md) // this will be a string 

const asyncMarked = new Marked({ async: true });
asyncMarked.parse(md) // this will be a Promise<string>
```

Reply:
maybe there's a way to have typed returned instance ?
the default instance, without any modification, will be synch, right ?

Reply:
Maybe. Typescript would have to run your application to see what the return type is.

Yes default without any extensions is sync